### PR TITLE
refactor: remove superfluous function-like macros

### PR DESCRIPTION
**Pull request by dundargoc**
Remove following macros:
CONVERTED
HMLL_ITER_BACK
PACK_STRUCT
IGNORE_BUF
find_shada_parameter
path_try_shorten_fname
incr
decr

### CODE REVIEW

File: src/nvim/mouse.c [hunk 1]
```diff
@@ -682,38 +682,39 @@ static int mouse_adjust_click(win_T *wp, int row, int col)
 
   vcol = offset;
 
-#define incr() nudge++; ptr_end += utfc_ptr2len(ptr_end)
-#define decr() nudge--; ptr_end -= utfc_ptr2len(ptr_end)
```
Comment (justinmk) [inline src/nvim/mouse.c:686]: think it's worth keeping these. They are just documentation. But they should be all uppercase.

Comment (dundargoc) [reply src/nvim/mouse.c:686]: What do you mean by documentation? It's just a macro function? I'm reverting the change now, nothing against it, I'm just confused :).

Comment (justinmk) [reply src/nvim/mouse.c:686]: Names/symbols can/should serve the same purpose as documentation.

Comment (dundargoc) [reply src/nvim/mouse.c]: OK, fixed :+1:

File: src/nvim/mouse.c [hunk 2]
```diff
@@ -682,39 +682,38 @@ static int mouse_adjust_click(win_T *wp, int row, int col)
 
   vcol = offset;
 
+#define incr() nudge++; ptr_end += utfc_ptr2len(ptr_end)
+#define decr() nudge--; ptr_end -= utfc_ptr2len(ptr_end)
```
Comment (justinmk) [inline src/nvim/mouse.c]: while we're here, they should be uppercase

```c
#define INCR() nudge++; ptr_end += utfc_ptr2len(ptr_end)
#define DECR() nudge--; ptr_end -= utfc_ptr2len(ptr_end)

Comment (dundargoc) [reply src/nvim/mouse.c]: Forgot. Thanks for the reminder.

**Comment by justinmk** [comment]
macos CI failure is a perf test, likely a one-off : 

```
2021-10-05T19:39:54.1528610Z 1 FAILED:
2021-10-05T19:39:54.1529060Z Found errors in Test_pattern_compile_speed():
2021-10-05T19:39:54.1530420Z function RunTheTest[37]..Test_pattern_compile_speed line 7: Expected range 0.01 - 10.0, but got 16.449491

**Pull request merged**